src/test/rustdoc-ui/automatic-links.rs

```diff
@@ -0,0 +1,58 @@
+#![deny(automatic_links)]
+
+/// [http://a.com](http://a.com)
+//~^ ERROR unneeded long form for URL
+/// [http://b.com]
+//~^ ERROR unneeded long form for URL
+///
+/// [http://b.com]: http://b.com
+///
+/// [http://c.com][http://c.com]
+pub fn a() {}
+
+/// https://somewhere.com
+//~^ ERROR this URL is not a hyperlink
+/// https://somewhere.com/a
+//~^ ERROR this URL is not a hyperlink
+/// https://www.somewhere.com
+//~^ ERROR this URL is not a hyperlink
+/// https://www.somewhere.com/a
+//~^ ERROR this URL is not a hyperlink
```

Comment:
You fixed the code, but it would be nice to have a test too.
```suggestion
/// https://www.somewhere.com/a
//~^ ERROR this URL is not a hyperlink
/// https://subdomain.example.com
//~^ ERROR not a hyperlink
```